### PR TITLE
AB#39440 stop query runner triggering forever

### PR DESCRIPTION
**Pull request by lhao03**
Fixed the bug of the query runner running forever.

To reproduce:
- go to `interns/dev`
- switch to app mode
- the run query button will run forever

found another bug:
image
image

`newPermissionModeType = PERMISSION_MODE_TYPE.TeamsApp;` was deleted.

`dispatch(toggleRSCPopup(query));` is causing the infinite run query.

### CODE REVIEW

**Review by ayubun — APPROVED** [review]
LGTM and the bugfixes mentioned are fixed for me c:
Notably, before merging:

When I run `https://graph.microsoft.com/v1.0/me` in non-logged-in mode, the query runner goes into the infinite trigger (likely the exact bug that was being fixed for app mode in this PR). I am wondering if this is a related issue to this exact PR, or if we should make a new ticket for it? If it's something that would take more than 1 min to add, I don't think it should be included and we can make a new bug ticket for it, but I just thought it would be useful to mention here ^^

**Comment by lhao03** [comment]
> When I run `https://graph.microsoft.com/v1.0/me` in non-logged-in mode, the query runner goes into the infinite trigger (likely the exact bug that was being fixed for app mode in this PR). I am wondering if this is a related issue to this exact PR, or if we should make a new ticket for it? If it's something that would take more than 1 min to add, I don't think it should be included and we can make a new bug ticket for it, but I just thought it would be useful to mention here ^^

Does that behavior also happen on the main graph explorer?

**Comment by ayubun** [comment]
> > When I run `https://graph.microsoft.com/v1.0/me` in non-logged-in mode, the query runner goes into the infinite trigger (likely the exact bug that was being fixed for app mode in this PR). I am wondering if this is a related issue to this exact PR, or if we should make a new ticket for it? If it's something that would take more than 1 min to add, I don't think it should be included and we can make a new bug ticket for it, but I just thought it would be useful to mention here ^^
> 
> Does that behavior also happen on the main graph explorer?

… Great q.. let me check lol

**Comment by ayubun** [comment]
> > When I run `https://graph.microsoft.com/v1.0/me` in non-logged-in mode, the query runner goes into the infinite trigger (likely the exact bug that was being fixed for app mode in this PR). I am wondering if this is a related issue to this exact PR, or if we should make a new ticket for it? If it's something that would take more than 1 min to add, I don't think it should be included and we can make a new bug ticket for it, but I just thought it would be useful to mention here ^^
> 
> Does that behavior also happen on the main graph explorer?

It doesn’t seem to be the case

File: src/app/services/actions/query-action-creators.ts [hunk 1]
```diff
@@ -128,17 +129,16 @@ export function toggleRSCPopup(query: IQuery): Function {
       return authenticatedRequest(dispatch, query).then(async (response: Response) => {
         const result = await parseResponse(response, respHeaders);
         for (const i of result.value) {
-          // check if Sample Teams App is installed
           if (i.teamsApp.externalId === TEAMS_APP_ID) {
```
Comment (omersayshi) [inline src/app/services/actions/query-action-creators.ts]: I know this wasnt part of your task, but is it possible to use a better variable name than i? Just for it to be easier to understand what each item in the list might be

**Review by omersayshi — APPROVED** [review]
Works like a charm and no apparent bugs, good job

**Comment by lhao03** [comment]
> > > When I run `https://graph.microsoft.com/v1.0/me` in non-logged-in mode, the query runner goes into the infinite trigger (likely the exact bug that was being fixed for app mode in this PR). I am wondering if this is a related issue to this exact PR, or if we should make a new ticket for it? If it's something that would take more than 1 min to add, I don't think it should be included and we can make a new bug ticket for it, but I just thought it would be useful to mention here ^^
> > 
> > 
> > Does that behavior also happen on the main graph explorer?
> 
> It doesn’t seem to be the case

I think you can make a bug ticket for this, because in user mode not logged in it shouldn't be making a call to check if rsc is installed so I think it would be a longer than 1 min fix

File: src/app/services/actions/query-action-creators.ts (hunk 1 above)
Comment (lhao03) [reply src/app/services/actions/query-action-creators.ts]: c7da6b2f138edd68423247c86ae7ac624f3f6875

**Pull request merged**